src/imap_mag/config/AppSettings.py

```diff
@@ -60,7 +61,7 @@ def settings_customise_sources(
         return (
             # Highest priority
             init_settings,
-            env_settings,
+            NestedAliasEnvSettingsSource(settings_cls),
```

Comment:
Ha thats just the obvious name I would have guessed to make this work. Not. 
Good find.